src/components/layouts/collection-page-layout.tsx

```diff
@@ -530,24 +535,84 @@ const CollectionPageLayout: React.FunctionComponent<CoursePageLayoutProps> = ({
                     }}
                   >
                     <div
-                      className="text-gray-600 dark:text-gray-300 flex flex-row items-center rounded hover:bg-gray-200 
-                      dark:hover:bg-gray-700 border-gray-300 bg-white dark:bg-gray-800 dark:border-gray-600 px-4 py-2 border transition-colors text-sm xs:text-base ease-in-out"
+                      className={
+                        ' flex flex-row items-center rounded  px-4 py-2 border transition-all text-sm xs:text-base ease-in-out duration-150 shadow-sm ' +
+                        (isFavorite
+                          ? 'hover:bg-blue-500 bg-blue-600 border-blue-700 text-white'
+                          : 'bg-white text-gray-600 hover:bg-gray-200 dark:text-gray-300 dark:hover:bg-gray-700 border-gray-300  dark:bg-gray-800 dark:border-gray-600')
+                      }
                     >
                       <BookmarkIcon
-                        className="w-4 h-4 mr-1"
+                        className="w-4 h-4 mr-2"
                         fill={isFavorite}
                       />{' '}
-                      Bookmark
+                      {isFavorite ? 'Bookmarked' : 'Bookmark'}
                     </div>
                   </button>
                 ) : (
-                  <div
-                    className="text-gray-600 dark:text-gray-300 flex flex-row items-center rounded hover:bg-gray-200 
-                  dark:hover:bg-gray-700 border-gray-300 bg-white dark:bg-gray-800 dark:border-gray-600 px-4 py-2 border transition-colors text-sm xs:text-base ease-in-out opacity-30"
+                  <DialogButton
+                    buttonText="Bookmark"
+                    title="Sign in to bookmark"
+                    buttonStyles="text-gray-600 dark:text-gray-300 flex flex-row items-center rounded hover:bg-gray-100 
+                      dark:hover:bg-gray-700 border-gray-300 bg-white dark:bg-gray-800 dark:border-gray-600 px-4 py-2 border transition-colors text-sm xs:text-base ease-in-out opacity-90 shadow-sm"
                   >
-                    <BookmarkIcon className="w-4 h-4 mr-1" /> Bookmark
-                  </div>
+                    <div className="flex flex-col space-y-4">
```

Comment:
suggestion [reuse]: We have a common login form that should probably be repurposed for this dialog

https://github.com/eggheadio/egghead-next/blob/03c48cb041dc7e0c83d9fcc2c31881e649dc09e4/src/pages/login.tsx

This form is exported from a "page" which is unorthodox, but is also used as a component in the "login cta overlay" that appears on lessons

https://github.com/eggheadio/egghead-next/blob/f120bf7a8a1390b3efddf6796bc698cb12bc5cdc/src/components/pages/lessons/overlay/email-capture-cta-overlay.tsx#L24-L41

The form is modestly customizable but would benefit from extraction into a component versus a page. This is somewhat arbitrary as a page is "just a component" but makes sense as we reuse it.

This might require some adjustments to the form component itself and/or the design presented here